temporal/api/cloud/namespace/v1/message.proto

```diff
@@ -124,3 +125,25 @@ message Namespace {
     // Will not be set if the namespace has never been modified.
     google.protobuf.Timestamp last_modified_time = 11;
 }
+
+enum FailoverType {
+    FAILOVER_TYPE_UNSPECIFIED = 0;
+    FAILOVER_TYPE_GRACEFUL = 1;
+    FAILOVER_TYPE_FORCEFUL = 2;
+}
+
+enum FailoverStatus {
+    FAILOVER_STATUS_UNSPECIFIED = 0;
+    FAILOVER_STATUS_SUCCEEDED = 1;
+    FAILOVER_STATUS_FAILED = 2;
+}
+
+message FailoverRecord {
+    FailoverType failover_type = 1;
+    temporal.api.cloud.region.v1.Region source_region = 2;
```

Comment:
Lets just use a string instead like the way we do on the namespace message. 
https://github.com/temporalio/api-cloud/blob/main/temporal/api/cloud/namespace/v1/message.proto#L48C4-L54C33

Reply:
same concern. If there is a plan to multi cloud. I think we should move to this Region object rather than use string region id.

Reply:
Multi cloud is addressed by the region id. We prefix the cloud type to the region. 

```
$ omni ocld cloud-apis regions list
{
	"regions": [
		{
			"id": "aws-us-east-1",
			"cloudProvider": "aws",
			"cloudProviderRegion": "us-east-1",
			"location": "US East (N. Virginia)"
		},
		{
			"id": "aws-us-west-2",
			"cloudProvider": "aws",
			"cloudProviderRegion": "us-west-2",
			"location": "US West (Oregon)"
		}
	]
}
```